### PR TITLE
CO-331 enforce provider queue truth

### PR DESCRIPTION
**Pull request by Kbediako**
## What

- Hold helper-created follow-up issues in `Backlog` while their Immediate Traceability packet and registry mirrors are still pending.
- Count queued retry and resumable provider claims as admission-cap occupants for `max_allowed` while avoiding self-counting during retry/resume relaunch.
- Align provider intake active summary identifiers with queued retry capacity, and add CO-331 docs/task mirrors plus regression coverage.

## Why

CO-331 captured live queue drift where follow-ups explicitly set to `Backlog` appeared as `Ready`, and `co-status` reported active issue identifiers above `max_allowed`. Retry/resumable work is active queue pressure even when it is not currently `running`.

## Validation

- `node scripts/delegation-guard.mjs`
- `node scripts/spec-guard.mjs --dry-run`
- `npm run build`
- `npm run lint` (passes with existing `DelegationMcpHealth.test.ts` no-explicit-any warnings)
- `npx vitest run orchestrator/tests/ProviderIntakeState.test.ts orchestrator/tests/ProviderIssueHandoff.test.ts orchestrator/tests/ProviderOperatorAutopilot.test.ts`
- `npm run test`
- `npm run docs:check`
- `npm run docs:freshness`
- `npm run repo:stewardship`
- `node scripts/diff-budget.mjs --base origin/main`
- `npm run pack:smoke`
- Manifest-backed `codex-orchestrator review --base origin/main`: clean success

## Summary by CodeRabbit

* **New Features**
  * Backlog promotion now blocks helper-created follow-ups until required traceability is present, surfacing an explicit hold reason.
  * Queued-retry and resumable claims count toward concurrent admission limits to prevent exceeding max capacity.
  * Provider intake, queue state and status surfaces are kept aligned with active issue identifiers and admission occupancy.

* **Tests**
  * Added tests covering backlog promotion holds, queued/resumable capacity counting, webhook/admission edge cases.

* **Documentation**
  * New PRD, tech spec, action plan, tasks and registry entries documenting the queue-cap contract and validation criteria.

### CODE REVIEW

File: orchestrator/src/cli/control/providerIssueHandoff.ts [hunk 1]
```diff
@@ -1927,7 +1933,10 @@ export function createProviderIssueHandoffService(
         claimWorkerHost: input.claim.worker_host ?? null,
         previousRun: input.previousRun ?? null
       });
-      const admissionGate = await createProviderAdmissionGate();
+      const admissionGate = await createProviderAdmissionGate({
+        excludeProviderKey: input.claim.retry_queued === true ? input.claim.provider_key : null,
+        preferredWorkerHost
+      });
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/control/providerIssueHandoff.ts:1942]: @coderabbitai Addressed in the current head `650ab01bb`: both start-flow `createProviderAdmissionGate` calls now pass only `excludeClaimProviderKey` and no `preferredWorkerHost`; resume keeps the host hint behavior separately. Please re-review/resolve.

**Comment by Kbediako** [comment]
Acknowledging current bot feedback for review handoff: CodeRabbit approved head `33ed37b81`; all prior actionable CodeRabbit threads are addressed/resolved, required checks are green, and the remaining CodeRabbit docstring-coverage warning/finishing-touch suggestions are non-blocking generated guidance for this diff. The Codex connector quota notice is also acknowledged; manifest-backed local standalone review completed successfully before handoff.

File: .agent/task/linear-9d8bb7c9-3ca1-422e-9cdf-ef90a4fbe7b7.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+## Checklist
+
+- [x] Docs-first packet created.
+- [ ] Backlog follow-up hold implemented.
+- [ ] Retry/resumable admission capacity implemented.
+- [ ] Provider intake summary projection aligned.
+- [ ] Focused regressions added.
+- [ ] Full validation and review gates completed.
+
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply .agent/task/linear-9d8bb7c9-3ca1-422e-9cdf-ef90a4fbe7b7.md:36]: @coderabbitai Acknowledged and addressed on current head `33ed37b81`: pre-implementation issue-quality/manual task-spec review gates are explicit in the checklist and review notes, and manifest-backed standalone review is tracked separately before handoff.

File: tasks/specs/linear-9d8bb7c9-3ca1-422e-9cdf-ef90a4fbe7b7.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+Restore one truthful queue contract across `Backlog`, `Ready`, `provider-intake-state.json`, `co-status`, `max_allowed`, active issue identifiers, resumable claims, and follow-up issue admission.
+
+## Target Contract
+
+- Helper-created follow-up issues that are explicitly placed in `Backlog` stay non-admitted until a real queue policy promotes them.
+- Resumable claims and queued retry claims occupy provider admission capacity.
+- `provider-intake-state.json` and `co-status` expose the same active issue identifiers used by admission.
+- Queue-cap or promotion drift is visible through concrete hold/block evidence.
+
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply tasks/specs/linear-9d8bb7c9-3ca1-422e-9cdf-ef90a4fbe7b7.md:23]: @coderabbitai Acknowledged and addressed on current head `33ed37b81`: the current/reference/target parity matrix is present in the task spec and preserves `Backlog`, `Ready`, `provider-intake-state.json`, `co-status`, active identifiers, queued retry, and resumable capacity semantics.

File: orchestrator/src/cli/control/providerIssueHandoff.ts [hunk 4]
```diff
@@ -1819,6 +1824,8 @@ export function createProviderIssueHandoffService(
       const workerHost = resolveRehydratedActiveRunWorkerHost(input.run, input.claim);
       const resumeWorkerHost = await resolveResumeWorkerHost(workerHost);
       const admissionGate = await createProviderAdmissionGate({
+        excludeClaimProviderKey:
+          input.claim.retry_queued === true ? input.claim.provider_key : null,
         forceLocalWorkerOnly: resumeWorkerHost === null,
         preferredWorkerHost: resumeWorkerHost
       });
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/control/providerIssueHandoff.ts:1833]: @coderabbitai Acknowledged and addressed on current head `33ed37b81`: resume admission now excludes the claim when `retry_queued === true` or `state === "resumable"`, preventing a plain resumable claim from occupying its own resume slot.

File: orchestrator/tests/ProviderIssueHandoff.test.ts [hunk 5]
```diff
@@ -7589,6 +7580,424 @@ describe('createProviderIssueHandoffService', () => {
...
+      }
+    });
+    expect(launcher.start).not.toHaveBeenCalled();
+    expect(launcher.resume).not.toHaveBeenCalled();
+    expect(warnSpy).toHaveBeenCalledWith(
+      expect.stringContaining(
+        `[provider-issue-run-discovery] skipping unreadable manifest ${brokenPaths.manifestPath}:`
+      )
+    );
+    vi.useRealTimers();
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/tests/ProviderIssueHandoff.test.ts:8001]: @coderabbitai Acknowledged and addressed on current head `33ed37b81`: the fake-timer test now uses `try/finally` cleanup for both `warnSpy.mockRestore()` and `vi.useRealTimers()`.
